client/engine/engine.go

```diff
@@ -230,8 +230,11 @@ func (e *Engine) attemptProgress(objective protocols.Objective) (outgoing Object
 func (e *Engine) getOrCreateObjective(message protocols.Message) (protocols.Objective, error) {
 	id := message.ObjectiveId
 
-	objective, ok := e.store.GetObjectiveById(id)
-	if !ok {
+	objective, err := e.store.GetObjectiveById(id)
+
+	if err == nil {
+		return objective, nil
+	} else if errors.Is(err, store.ErrNoSuchObjective) {
```

Comment:
I prefer `ok` over having to check errors but maybe that's an ok trade off for `GetObjectiveById` being able to return various error types.